src/ray/rpc/raylet/raylet_client.cc

```diff
@@ -0,0 +1,385 @@
+
+#include "src/ray/rpc/raylet/raylet_client.h"
+
+namespace ray {
+namespace rpc {
+
+RayletClient::RayletClient(const std::string &raylet_socket, const WorkerID &worker_id,
+                           bool is_worker, const JobID &job_id, const Language &language,
+                           int port)
+    : worker_id_(worker_id),
+      is_worker_(is_worker),
+      job_id_(job_id),
+      language_(language),
+      port_(port),
+      main_service_(),
+      work_(main_service_),
+      client_call_manager_(main_service_),
+      heartbeat_timer_(main_service_) {
+  std::shared_ptr<grpc::Channel> channel =
+      grpc::CreateChannel("unix://" + raylet_socket, grpc::InsecureChannelCredentials());
+  stub_ = RayletService::NewStub(channel);
+
+  rpc_thread_ = std::thread([this]() { main_service_.run(); });
+  RAY_LOG(DEBUG) << "Connecting to unix socket: "
+                 << "unix://" + raylet_socket
+                 << ", is worker: " << (is_worker_ ? "true" : "false")
+                 << ", worker id: " << worker_id;
+  // Try to register client for 10 times.
```

Comment:
```suggestion
  // Try to register client 10 times.
```

Can we make this a named constant or configurable somewhere? It's easy for magic numbers like these to get lost.